### DISCUSSION TITLE
Control Plane Deployment Instructions

### DISCUSSION THREAD

**Discussion by walski** [ubicloud/ubicloud#2364] [Q&A]
Hey,

are there any resources / docs out there how to deploy a control plane in production beyond the 3 bullet points in the [Build Your Own Cloud](https://www.ubicloud.com/docs/quick-start/build-your-own-cloud) docs that get a control plane up and running locally?

Specifically I'd love to get an idea on if there is any easy way to run the control plan in a fail-safe/high-available way? And what are the suggestions in terms of RAM, CPU power to run a control plane instance?

Thanks for your help!

Thorben

**Comment by ozgune** [comment]
Hi @walski - thank you for the question!

We focused on releasing new cloud services and features so far on our managed offering so far. That sadly meant that we haven't spent enough time on making Ubicloud's self-deploy steps production ready.

We discuss this topic every other month. When there's enough demand from the community or we're at a natural break point with new cloud services, we'll prioritize this.

Until then, I'm going to upvote your question. I also had two questions as input into our product roadmap.

* How big of a Ubicloud deployment do you have in mind? A few servers, dozens?
* Which Ubicloud services would you like to use? VMs, block storage, load balancer, managed Postgres, ABAC, or AI endpoints

Thank you again for the question & your understanding.

**Comment by walski** [comment]
Hey, thanks for the reply :) Looking forward to any progress on this point.

To your questions:

We are looking into a replacement for our production setup which currently involves 3 larger servers. But as part of that migration I wouldn't be surprised if we expand that setup by a few machines. We would probably stay between 5 and 10 servers, though.

That said: We are currently kind of limited in terms of what we do by the way our infra setup works (a bunch of custom Ansible roles). If we would have a setup that e.g. would allow to easily add new pg clusters etc, I can see a future where we would leverage that e.g. with a more sophisticated staging or CI deployments of our product etc. In the end, servers are relatively cheap. Managing them "correctly" is what is expensive to us atm. (And we can't use a public cloud due to data privacy / data security concerns of some of our customers).

We'd use VMs, block storage, load balancer, and managed postgres. If there is managed Redis, we would also love to use that. That said: Postgres would be the main deciding factor for us as managing a high available cluster ourselves is just a nightmare and "outsourcing" that would be amazing.

Last but not least: We would also totally be down using a managed/hosted control plane, as long as the worker nodes could be fully under our control and the control plane would not suffer from not being physically close to the workers.

**Reply by ukwuezeobinna** [reply]
Any update on this?
